code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/download/RulesLoader.java

```diff
@@ -104,6 +104,7 @@ public void loadFromUrl(
         final NetworkCallback networkCallback =
                 response -> {
                     final RulesLoadResult result = handleDownloadResponse(url, response);
```

Comment:
response can be null and currently `handleDownloadResponse` does not also handle null case. Can you fix this callback overall to handle null case?